### PR TITLE
Add validator for OAuth 

### PR DESCRIPTION
**Pull request by psantos1113**
Was going to add this in 6.2.0 but the warnings are treated as errors and would prevent saving.  This will allow pushing the save and test button and having the UI display a warning message.

Fixed the cause of a circular dependency in the ConfigurationFieldModelConverter.  The class uses the descriptor keys.  If we use the DescriptorMap it needs all the descriptors which is where the validator is getting used.  So it can't autowire the objects correctly.

### CODE REVIEW

File: src/main/java/com/synopsys/integration/alert/channel/azure/boards/AzureRedirectUtil.java [hunk 1]
```diff
@@ -38,6 +38,10 @@ public AzureRedirectUtil(AlertProperties alertProperties) {
         this.alertProperties = alertProperties;
     }
 
+    /**
+     * The OAuth callback controller will redirect back to the Alert UI and is the only place where this method.
```
Comment (bamandel) [inline src/main/java/com/synopsys/integration/alert/channel/azure/boards/AzureRedirectUtil.java:42]: Incomplete sentence?

File: src/main/java/com/synopsys/integration/alert/channel/azure/boards/oauth/AzureOAuthTokenValidator.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    @Autowired
+    public AzureOAuthTokenValidator(AzureRedirectUtil azureRedirectUtil, AzureBoardsCredentialDataStoreFactory azureBoardsCredentialDataStoreFactory,
+        ConfigurationFieldModelConverter configurationFieldModelConverter, ProxyManager proxyManager) {
+        this.azureRedirectUtil = azureRedirectUtil;
+        this.azureBoardsCredentialDataStoreFactory = azureBoardsCredentialDataStoreFactory;
+        this.configurationFieldModelConverter = configurationFieldModelConverter;
+        this.proxyManager = proxyManager;
+    }
+
+    public ValidationResult validate(FieldValueModel fieldValueModel, FieldModel fieldModel) {
```
Comment (bamandel) [inline src/main/java/com/synopsys/integration/alert/channel/azure/boards/oauth/AzureOAuthTokenValidator.java:57]: is/will this method be used? I'm not really sure what the point of it is.

Comment (psantos1113) [reply src/main/java/com/synopsys/integration/alert/channel/azure/boards/oauth/AzureOAuthTokenValidator.java:57]: I can remove it now that I've figured out the circular dependency issue.

File: src/main/java/com/synopsys/integration/alert/channel/azure/boards/AzureRedirectUtil.java (hunk 1 above)
Comment (psantos1113) [reply src/main/java/com/synopsys/integration/alert/channel/azure/boards/AzureRedirectUtil.java:42]: Thanks I will add that in.

**Pull request merged**